File: integraedSW/log_example/src/log_example.cpp
#include <cstdio>
#include <unistd.h>


#include "rclcpp/rclcpp.hpp"



int main(int argc, char ** argv)
{
  (void) argc;
  (void) argv;

  rclcpp::init(argc, argv);

  auto node = std::make_shared<rclcpp::Node>("rclcpp_log_Test");

  while(1){
  RCLCPP_DEBUG(node->get_logger(), "test log debug");
  sleep(1);
  RCLCPP_INFO(node->get_logger(), "test log info");
  sleep(1);
  RCLCPP_WARN(node->get_logger(), "test log warn");
  sleep(1);
  RCLCPP_ERROR(node->get_logger(), "test log error");
  sleep(1);
  RCLCPP_FATAL(node->get_logger(), "test log fatal");
  sleep(1);}







return 0;
}

File: integraedSW/log_example/src/log_example.py
## Log Test

'''
DEBUG = 10
ERROR = 40
FATAL = 50
INFO = 20
UNSET = 0
WARN = 30


You can get this enum from rqt /rosout topic
'''



import rclpy
import time

rclpy.init()
node = rclpy.create_node("rclpy_log_test")

while(1):
    node.get_logger().info("Info flagged")
    time.sleep(1)
    node.get_logger().debug("debug flagged")
    time.sleep(1)
    node.get_logger().warning("warning flagged")
    time.sleep(1)
    node.get_logger().error("error flagged")
    time.sleep(1)
    node.get_logger().fatal("fatal flagged")
    time.sleep(1)
